src/wp-includes/rest-api.php

```diff
@@ -1305,6 +1305,24 @@ function rest_validate_value_from_schema( $value, $args, $param = '' ) {
 			/* translators: 1: Parameter, 2: Number. */
 			return new WP_Error( 'rest_invalid_param', sprintf( __( '%1$s must contain at most %2$s items.' ), $param, number_format_i18n( $args['maxItems'] ) ) );
 		}
+
+		if ( isset( $args['uniqueItems'] ) && $args['uniqueItems'] ) {
+			foreach ( $value as $index => &$object ) {
+				if ( is_array( $object ) ) {
+					ksort( $object );
```

Comment:
Does this work for deeply nested objects?

I think we should extract a `rest_is_array_unique` function.